Add checks for theme app extension block js/css (#334)
Shopify/theme-check: docs/checks/asset_size_app_block_javascript.md

# Prevent Abuse on Server Rendered App Blocks (`AssetSizeAppBlockJavaScript`)

For server rendered app blocks, it is an anti-pattern to execute large JavaScript bundles on every page load

This doesn't mean they don't have a reason to exist. For instance, chat widgets are mini applications embedded inside web pages. Designing such an app with server rendered updates would be absurd. However, if only 10% of the users interact with the chat widget, the other 90% should not have to execute the entire bundle on every page load.

The natural solution to this problem is to implement the chat widget using the [Import on Interaction Pattern][ioip].

## Check Details

This rule disallows the use of block JavaScript files and external scripts to have a compressed size greater than a configured `threshold_in_bytes`.

:-1: Examples of **incorrect** code for this check:
```liquid
<!-- Here assets/chat-widget.js is more than 10KB gzipped. -->
{% schema %}
{
  ...

>   "javascript": "chat-widjet.js"

Reviewer: ```suggestion
  "javascript": "chat-widget.js"
```

}
{% endschema %}
```

## Check Options

> The default configuration is the following.

Reviewer: ```suggestion
The default configuration is the following:
```

```yaml
AssetSizeAppBlockJavaScript:
  enabled: true
  threshold_in_bytes: 10000
```

### `threshold_in_bytes`

The `threshold_in_bytes` option (default: `10000`) determines the maximum allowed compressed size in bytes that a single JavaScript file can take.

This includes theme and remote scripts.

## When Not To Use It

When you can't do anything about it, it is preferable to disable this rule using the comment syntax:

```
{% comment %}theme-check-disable AppBlockAssets{% endcomment %}
<script src="https://code.jquery.com/jquery-3.6.0.min.js" defer></script>

> {% comment %}theme-check-enable AppBlockAssets{% endcomment %}

Reviewer: I feel like this snippet doesn't apply here? Or that the example is the wrong one.

```

This makes disabling the rule an explicit affair and shows that the code is smelly.

## Version

> This check has been introduced in Theme Check 1.1.0

Reviewer: ```suggestion
This check has been introduced in THEME_CHECK_VERSION
```

We have a releasing step that changes those automatically for us. This way we don't risk releasing it in 1.0.1 and have the wrong number.

Author: perfect, I was wondering what I should do here!
